### PR TITLE
HIPP-1548: Wire up cancel prod access request

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  def submitRequest(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      validate(request).fold(
+        call => Future.successful(Redirect(call)),
+        data => {
+          val accessRequests = getCancellableRequests(data)
+          Future.sequence(getCancellableRequests(data).map(accessRequest => apiHubService.cancelAccessRequest(accessRequest.id, request.user.email)))
+            .flatMap(_ => sessionRepository.clear(request.user.userId))
+            .flatMap(_ => Future.successful(Ok(cancelAccessRequestSuccessView(data.application, Some(request.user), accessRequests))))
```
Comment (victorarbuesmallada) [inline app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala:59]: can this be a `map` ?
```suggestion
            .map(_ => Ok(cancelAccessRequestSuccessView(data.application, Some(request.user), accessRequests)))
```

Comment (llmikeyj) [reply app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala:59]: I don't think so, no. That changes the return type

[error]    |Found:    scala.concurrent.Future[scala.concurrent.Future[play.api.mvc.Result] |
[error]    |Required: scala.concurrent.Future[play.api.mvc.Result]

File: app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      apis <- validateApis(request.userAnswers)
+      _ <- validateDeclaration(request.userAnswers)
+    } yield Data(application, accessRequests, apis)
+  }
+
+  private def validateApplication(userAnswers: UserAnswers): Either[Call, Application] = {
+    userAnswers.get(CancelAccessRequestApplicationPage) match {
+      case Some(application) => Right(application)
+      case None => Left(controllers.routes.JourneyRecoveryController.onPageLoad())
+    }
```
Comment (victorarbuesmallada) [inline app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala:88]: Not sure if you would prefer this:
```suggestion  userAnswers.get(CancelAccessRequestApplicationPage).toRight(controllers.routes.JourneyRecoveryController.onPageLoad())
```

**Review by victorarbuesmallada — DISMISSED** [review]
LGTM, small comments

File: app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala (hunk 2 above)
Comment (llmikeyj) [reply app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala:88]: Thanks for the suggestion. I haven't used this style before. We generally use the match case pattern across the hub though, and personally I prefer that style as i find the intent is clearer?

File: app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala (hunk 1 above)
Comment (victorarbuesmallada) [reply app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala:59]: oh I mean, we shouldn't return a `Future` inside the `map` but just a `Result`:
```
.flatMap(_ => Future(_ => Ok(...
```
to
```
.map(_ => Ok(...
```

Comment (llmikeyj) [reply app/controllers/application/cancelaccessrequest/CancelAccessRequestEndJourneyController.scala:59]: Ah ok

**Review by PaulCDurham — APPROVED** [review]

**Pull request merged**